toolchain/sem_ir/inst_namer.cpp

```diff
@@ -487,16 +501,24 @@ auto InstNamer::CollectNamesInBlock(ScopeId top_scope_id,
         continue;
       }
       case CARBON_KIND(AssociatedEntityType inst): {
-        // TODO: Try to get the name of the interface associated with
-        // `inst.interface_type_id`.
-        if (auto fn_ty =
-                sem_ir_->types().TryGetAs<FunctionType>(inst.entity_type_id)) {
-          add_inst_name_id(sem_ir_->functions().Get(fn_ty->function_id).name_id,
-                           ".assoc_type");
-        } else {
-          // TODO: Handle other cases.
-          add_inst_name("assoc_type");
+        auto facet_type =
+            sem_ir_->types().TryGetAs<FacetType>(inst.interface_type_id);
+        if (!facet_type) {
+          // Should never happen, but we don't want the instruction namer to
```

Comment:
Log?

Reply:
We don't have a general logging infrastructure, and writing to stderr from here would be surprising.

How's this? We'll at least leave a marker in the SemIR that something went wrong.